site/data/en-US/docs/migration-guide.mdx

```diff
@@ -6,6 +6,316 @@ title: Migration Guide
+  );
+};
+```
+
+**3. WalletConnect legacy deprecation**
+
+RainbowKit no longer supports WalletConnect legacy parameters. Here is how you pass in custom WalletConnect parameters.
+
+Example with `connectorsForWallets`
+
+```diff
+import {
+  rainbowWallet,
+  coinbaseWallet,
+  metaMaskWallet,
+  walletConnectWallet,
+} from '@rainbow-me/rainbowkit/wallets';
+
+const connectors = connectorsForWallets({
+  projectId,
+  walletList: [
+    {
+      groupName: 'Recommended',
+      wallets: [
+        rainbowWallet,
+        coinbaseWallet,
+        metaMaskWallet,
+        walletConnectWallet,
+      ],
+    },
+  ],
+  appName,
++ walletConnectParameters: { ... };
+});
+```
+
+`RainbowKitWalletConnectParameters` type will help you to see what parameters you should pass in.
+
+**4. Custom wallet**
+
+If you've previously created your own wallet with RainbowKit you would need to update a few things.
+
+Here is example of a custom injected wallet.
+
+```diff
++ import { createConnector as wagmiCreateConnector } from 'wagmi';
++ import { injected } from 'wagmi/connectors';
+
+const injectedWallet = (): Wallet => ({
+  id: 'injected',
+  name: 'Injected Wallet',
+  iconUrl: async () => (await import('./injectedWallet.svg')).default,
+  iconBackground: '#fff',
++  mobile?: { ... };
++  desktop?: { ... };
++  qrCode?: { ... };
++  extension?: { ... };
+-  createConnector: () => ({
+-    connector: new InjectedConnector({
+-      chains,
+-      options,
+-    }),
+-  mobile?: { ... };
+-  desktop?: { ... };
+-  qrCode?: { ... };
+-  extension?: { ... };
+-  }),
++  createConnector: (walletDetails: WalletDetailsParams) => {
++    return wagmiCreateConnector((config) => ({
++      ...injected({
++        target: () => ({
++          id: "windowProvider",
++          name: "Window Provider",
++          provider: window.ethereum,
++        }),
++      })(config),
++      ...walletDetails,
++    }));
++  },
+});
+```
+
+Here is another example of a custom WalletConnect wallet.
+
+```diff
++ import { createConnector as wagmiCreateConnector } from 'wagmi';
++ import { walletConnect } from 'wagmi/connectors';
+
+const customWallet = (): Wallet => ({
+  id: 'custom',
+  name: 'Custom Wallet',
+  iconUrl: async () => (await import('./customWallet.svg')).default,
+  iconBackground: '#fff',
++  mobile?: { ... };
++  desktop?: { ... };
++  qrCode?: { ... };
++  extension?: { ... };
+-  createConnector: () => ({
+-    connector: new WalletConnectConnector({
+-      projectId: "...",
+-    }),
+-  mobile?: { ... };
+-  desktop?: { ... };
+-  qrCode?: { ... };
+-  extension?: { ... };
+-  }),
++  createConnector: (walletDetails: WalletDetailsParams) => {
++    return wagmiCreateConnector((config) => ({
++      ...walletConnect({
++        projectId: "...",
++      })(config),
++      ...walletDetails,
++    }));
++  },
+});
+```
+
+**5. Custom chains**
+
+We no longer support custom chains. You no longer can pass `chains` into `<RainbowKitProvider>`
```

Comment:
I think this should be stated more clearly - we can still provide custom chains via wagmi config while there is no need to pass into here.

Also I found that custom`iconUrl` is no longer supported?

Reply:
Yeah, just curious, without `iconUrl`, how will we include icons for chains that's are not built-in in rainbowkit?

![CleanShot-AFdaZkqR@2x](https://github.com/rainbow-me/rainbowkit/assets/126733611/5905c665-0ce7-46f2-9665-96baed6ef28f)

Reply:
Good catch! We'll introduce a custom api `getChains` to provide custom background color and icon url